pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -851,6 +851,9 @@ type S3Bucket struct {
 	// Bucket is the name of the bucket
 	Bucket string `json:"bucket"`
 
+	// Bucket secret reference
+	BucketSecret apiv1.SecretKeySelector `json:"bucketSecret"`
```

Comment:
probably should have `json:"bucketSecret,emitempty"` - otherwise this is required, also bucket should have `omitempty`

Reply:
probably should be `*apiv1.SecretKeySelector` than `apiv1.SecretKeySelector`